rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -0,0 +1,1424 @@
+            else:
+                output_length = LABEL_RANKING_LENGTH
+
+            ranking = list(zip(list(label_ids), message_sim))
+            ranking = ranking[:output_length]
+            label_ranking = [
+                {"name": self.inverted_label_dict[label_idx], "confidence": score}
+                for label_idx, score in ranking
+            ]
+
+        return label, label_ranking
+
+    def _predict_entities(
+        self, predict_out: Optional[Dict[Text, tf.Tensor]], message: Message
+    ) -> List[Dict]:
+        if predict_out is None:
+            return []
+
+        # load tf graph and session
+        predictions = predict_out["e_ids"].numpy()
+
+        tags = [self.inverted_tag_dict[p] for p in predictions[0]]
+
+        if self.component_config[BILOU_FLAG]:
+            tags = bilou_utils.remove_bilou_prefixes(tags)
+
+        entities = self._convert_tags_to_entities(
+            message.text, message.get("tokens", []), tags
+        )
+
+        extracted = self.add_extractor_name(entities)
+        entities = message.get("entities", []) + extracted
+
+        return entities
+
+    @staticmethod
+    def _convert_tags_to_entities(
+        text: Text, tokens: List[Token], tags: List[Text]
+    ) -> List[Dict[Text, Any]]:
+        entities = []
+        last_tag = "O"
+        for token, tag in zip(tokens, tags):
+            if tag == "O":
+                last_tag = tag
+                continue
+
+            # new tag found
+            if last_tag != tag:
+                entity = {
+                    "entity": tag,
+                    "start": token.start,
+                    "end": token.end,
+                    "extractor": "DIET",
+                }
+                entities.append(entity)
+
+            # belongs to last entity
+            elif last_tag == tag:
+                entities[-1]["end"] = token.end
+
+            last_tag = tag
+
+        for entity in entities:
+            entity["value"] = text[entity["start"] : entity["end"]]
+
+        return entities
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        """Return the most likely label and its similarity to the input."""
+
+        out = self._predict(message)
+
+        if self.component_config[INTENT_CLASSIFICATION]:
+            label, label_ranking = self._predict_label(out)
+
+            message.set("intent", label, add_to_output=True)
+            message.set("intent_ranking", label_ranking, add_to_output=True)
+
+        if self.component_config[ENTITY_RECOGNITION]:
+            entities = self._predict_entities(out, message)
+
+            message.set("entities", entities, add_to_output=True)
+
+    def persist(self, file_name: Text, model_dir: Text) -> Dict[Text, Any]:
+        """Persist this model into the passed directory.
+
+        Return the metadata necessary to load the model again.
+        """
+
+        if self.model is None:
+            return {"file": None}
```

Comment:
should we add warning?

Reply:
we already have a lot warnings during training, I don't think it'll be clear for a user during persist

Reply:
But that's an error when we are trying to persist something which isn't there, correct? So this shouldn't happen in the first place.

Reply:
no, it's not an error. This happens when `ResponseSelector` is present, but there were no training data for it